.github/workflows/ci.yml

```diff
@@ -0,0 +1,96 @@
+name: ci-workflow
+
+on:
+  workflow_dispatch:
+    inputs:
+      repo:
+        description: 'Repository info'
+        required: true
+      branch:
+        description: 'Name of the branch'
+        required: true
+      target_repository:
+        description: 'Repository name of the terratest'
+        required: true
+      target_pr:
+        description: 'PR number on the terratest repo'
+        required: true
+
+jobs:
+  ci-job:
+    runs-on: [ubuntu-latest]
+    steps:
+      - name: checkout to repo
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.event.inputs.repo }}
+          ref: ${{ github.event.inputs.branch }}
+      - name: list module files to lint
+        run: |
+          [ -d "modules/azure" ] && ls -li "modules/azure"
+      - name: lint modules/azure folder
+        id: azure_module_lint
+        uses: golangci/golangci-lint-action@v2.2.0
+        with:
+          version: v1.30
+          working-directory: modules/azure
+      - name: list test files to lint
+        run: |
+          [ -d "test/azure" ] && ls -li "test/azure"
+      - name: lint test/azure folder
+        id: azure_test_lint
+        uses: golangci/golangci-lint-action@v2.2.0
+        with:
+          version: v1.30
+          working-directory: test/azure
+      - name: login to azure cli
+        uses: azure/login@v1.1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+      - name: run go test for azure
+        id: azure_test
+        env:
+          AZURE_CREDENTIALS: ${{ secrets.AZURE_CREDENTIALS }}
+        run: |
+          az account show
```

Comment:
Sanity check: this doesn't show any secrets or confidential info, does it?

Reply:
This will not show secrets.  The secrets for subscription access are stored in github secrets so the log should mask out the values.

Reply:
that's correct, in the logs and Settings -> Secrets page, you can't see the secrets, it's always masked out.